roundrobin/stickysessions.go

```diff
@@ -81,15 +80,29 @@ func (s *StickySession) StickBackend(backend *url.URL, w *http.ResponseWriter) {
 	http.SetCookie(*w, cookie)
 }
 
-func (s *StickySession) isBackendAlive(needle *url.URL, haystack []*url.URL) bool {
+func (s *StickySession) isBackendAlive(needle string, haystack []*url.URL) (bool, *url.URL) {
 	if len(haystack) == 0 {
-		return false
+		return false, nil
 	}
 
-	for _, serverURL := range haystack {
-		if sameURL(needle, serverURL) {
-			return true
+	switch {
+	case strings.HasPrefix(needle, "http"):
```

Comment:
Checking the `http` prefix might not be enough as this would not cover `ws` and `wss`. There might also be some collision, especially with the `ws` scheme. What do you think about doing something like `key:hash`, this way you could just check for the `key:` prefix without limitations on the scheme?

Reply:
```go
	case strings.Contains(needle, "://"):
```

WDYT ?

Reply:
This sounds good either :+1: